src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ComCaseloadSearchService.kt

```diff
@@ -146,24 +135,14 @@ class ComCaseloadSearchService(
     )
   }
 
-  private fun List<PrisonerSearchPrisoner>.findBookingsWithHdc(): List<Long> {
-    val bookingsWithHdc = this
-      .filter { it.homeDetentionCurfewEligibilityDate != null }
-      .mapNotNull { it.bookingId?.toLong() }
-    val hdcStatuses = prisonApiClient.getHdcStatuses(bookingsWithHdc)
-    return hdcStatuses.filter { it.approvalStatus == "APPROVED" }.mapNotNull { it.bookingId }
-  }
-
-  private fun List<FoundProbationRecord>.filterOutHdc(prisonerRecords: Map<String, PrisonerSearchPrisoner>): List<FoundProbationRecord> {
-    if (prisonerRecords.isEmpty()) {
+  private fun List<FoundProbationRecord>.filterOutHdc(prisoners: Map<String, PrisonerSearchPrisoner>): List<FoundProbationRecord> {
+    if (prisoners.isEmpty()) {
       return this
     }
-    val prisonersForHdcCheck =
-      this.filter { it.licenceStatus == LicenceStatus.NOT_STARTED }.mapNotNull { prisonerRecords[it.nomisId] }
-    val bookingIdsWithHdc = prisonersForHdcCheck.findBookingsWithHdc()
+    val prisonersForHdcCheck = this.filter { it.licenceStatus == NOT_STARTED }.mapNotNull { prisoners[it.nomisId] }
 
-    val prisonersWithoutHdc = prisonerRecords.values.filterNot { bookingIdsWithHdc.contains(it.bookingId?.toLong()) }
-    return this.filter { it.isOnProbation == true || prisonersWithoutHdc.any { prisoner -> prisoner.prisonerNumber == it.nomisId } }
+    val hdcStatuses = hdcService.getHdcStatus(prisonersForHdcCheck)
+    return this.filter { it.isOnProbation == true || hdcStatuses.canBeSeenByCom(it.kind, it.bookingId!!) }
```

Comment:
I added booking id to FoundProbationRecord so we don't need to "downgrade" to using prison number